ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -160,6 +212,15 @@ public void updateNodeReceiveMetric(String hostName, int port, long bytesReceive
     nodeMetric.bytesReceivedLatency.update(timeTakenToReceiveInMs);
   }
 
+  /**
+   * Updates a few metrics when a request is timed-out without successful checkout of a connection
+   * @param connectionCheckOutAttempts number of times, connection check out was attempted before timing out
+   */
+  void updateConnectionTimedOutMetrics(int connectionCheckOutAttempts) {
```

Comment:
perhaps it would be better to update this directly rather than via this method as this seems like a one-off thing.